htdocs/core/class/notify.class.php

```diff
@@ -266,6 +269,20 @@ function send($notifcode, $object)
         else $sql.= " AND a.code = '".$notifcode."'";	// New usage
         $sql .= " AND s.rowid = ".$object->socid;
 
+		// Check notification per user
+        $sql.= "\nUNION\n";
+		$sql.= "SELECT  1 as user, c.email, c.rowid as cid, c.lastname, c.firstname, '$langs->defaultlang' as default_lang,";
+		$sql.= " a.rowid as adid, a.label, a.code, n.rowid, n.type";
+        $sql.= " FROM ".MAIN_DB_PREFIX."user as c,";
+        $sql.= " ".MAIN_DB_PREFIX."c_action_trigger as a,";
+        $sql.= " ".MAIN_DB_PREFIX."notify_def as n,";
+        $sql.= " ".MAIN_DB_PREFIX."element_contact as ec";
+        $sql.= " WHERE n.fk_user = c.rowid AND a.rowid = n.fk_action";
+        $sql.= " AND n.fk_user = ec.fk_socpeople";
```

Comment:
1 - I assumed that customer contacts and our internal contacts for common
objects need to be notified in the same way.

2- fk_user of table notify_def is used in the select query :

$sql.= " WHERE *n.fk_user = c.rowid* AND a.rowid = n.fk_action";


2016-05-05 19:40 GMT+01:00 Laurent Destailleur <notifications@github.com>:

> In htdocs/core/class/notify.class.php
> <https://github.com/Dolibarr/dolibarr/pull/4808#discussion_r62233651>:
>
> > @@ -266,6 +269,20 @@ function send($notifcode, $object)
> >          else $sql.= " AND a.code = '".$notifcode."'";	// New usage
> >          $sql .= " AND s.rowid = ".$object->socid;
> >
> > +		// Check notification per user
> > +        $sql.= "\nUNION\n";
> > +		$sql.= "SELECT  1 as user, c.email, c.rowid as cid, c.lastname, c.firstname, '$langs->defaultlang' as default_lang,";
> > +		$sql.= " a.rowid as adid, a.label, a.code, n.rowid, n.type";
> > +        $sql.= " FROM ".MAIN_DB_PREFIX."user as c,";
> > +        $sql.= " ".MAIN_DB_PREFIX."c_action_trigger as a,";
> > +        $sql.= " ".MAIN_DB_PREFIX."notify_def as n,";
> > +        $sql.= " ".MAIN_DB_PREFIX."element_contact as ec";
> > +        $sql.= " WHERE n.fk_user = c.rowid AND a.rowid = n.fk_action";
> > +        $sql.= " AND n.fk_user = ec.fk_socpeople";
>
> fk_socpeople can contains both id of a contact (llx_socpeople) and id of
> user (llx_user).
> If you make this link, you may report notification to user that does not
> ask it just because a contact was set for it.
> For notification "per user", you must use the field "fk_user" of table
> notify_def like we used the field fk_soc for notification to thirdparties.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Dolibarr/dolibarr/pull/4808/files/23c0a0a64e3e82e2bc8359d4a976d2c9b4f96f21#r62233651>
>



-- 
Bahfir Abbes


*Dolibarr preferred Partner.
<https://wiki.dolibarr.org/index.php/List_of_Dolibarr_partners_and_providers#Algeria_.2F_Alg.C3.A9rie>*
*Dolipar.org <http://Dolipar.org>*